File: JInstantSearch/app/src/main/java/com/woowrale/mvp/ui/search/home/SearchPresenter.java
package com.woowrale.mvp.ui.search.home;

import com.woowrale.mvp.ui.base.BasePresenter;

public class SearchPresenter extends BasePresenter<SearchView> {

    private SearchActivity searchActivity;

    public SearchPresenter(SearchActivity searchActivity) {
        this.searchActivity = searchActivity;
    }

    public void navigateToSearch(String searchType){
        if (searchType.equals("L")){
            searchActivity.navigateToSearchLocal();
        }else{
            searchActivity.navigateToSearchRemote();
        }
    }
}

File: KInstantSearch/app/src/main/java/com/woowrale/mvp/ui/search/home/SearchPresenter.kt
package com.woowrale.mvp.ui.search.home

import com.woowrale.mvp.ui.base.BasePresenter

class SearchPresenter(private val searchActivity: SearchActivity) : BasePresenter<SearchView>() {

    fun navigateToSearch(searchType: String) {
        if (searchType == "L") {
            searchActivity.navigateToSearchLocal()
        } else {
            searchActivity.navigateToSearchRemote()
        }
    }
}